interview-questions.md

```diff
@@ -9,41 +9,43 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+  Researched answer: React is a JavaScript library for building user interfaces. React makes it easy to efficiently update and render changes within your code. Also, React uses the virtual DOM which represents a real DOM that is manipulated by the browser, which allows the render of a webpage to be lightning fast. React uses JSX, which is a syntax extension to JavaScript. Users would want to use React, because of it’s quickness to render visual user interfaces out of the box.
 
-  Researched answer:
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: The render() method is required in order for the React class to run.
 
-  Researched answer:
+  Researched answer: The lifecycle method is broken down into 3 steps: mounting, updating, and unmounting. The mounting phase is a part of the constructor, render and methods. The update step will include the setState gets triggered. Lastly, the unmounting occurs when a component is being removed from the DOM. The lifecycle method that is required is the render method in the mounting phase.
 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is a JavaScript extension that is used in React. It allows JavaScript syntax to be used very easily in React code by using curly braces. The difference between HTML and JSX is JSX can only return 1 item. If there needs to be multiple items returned in JSX, there needs to be a fragment tag before and after all the JSX code. HTML code can return multiple items.
 
-  Researched answer:
+  Researched answer: JSX is a syntax extension of JavaScript that makes it easy to write HTML in React. One syntax difference between HTML and JSX is JSX self-closes the tags. An example of HTML is <div></div>, but JSX self closes itself like this: <div />. If HTML code is placed in React like the example above, it will not run. However, using the self-closing tag will run as expected.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is an extension added to a codebase. A feature of yarn is to add Jest, which is a testing framework to ensure the codebase is functioning correctly.
 
-  Researched answer:
+  Researched answer: Yarn is a JavaScript package manager that replaces the npm client or other package managers. The purpose of the package manager is to install a piece of code that serves a purpose. When 'yarn start' is ran, it runs the application in development mode and opens a local server. The local server can be opened in: http://localhost:3000.
```

Comment:
Make sure to address all parts of the question.  I don't see anything about what file(s) are modified when running yarn.